src/components/tabset/tabset.component.spec.ts

```diff
@@ -116,12 +121,56 @@ describe('Tabset Component', () => {
         expect(component.onTabDeselect).toHaveBeenCalled();
     });
 
+    it('should change to new tab when tab is programmatically selected by tab index', () => {
+
+        const tab1 = getTabForClass(0, nativeElement);
+        const tab3 = getTabForClass(2, nativeElement);
+
+        // check first tab is active
+        expect(tab1.classList.contains('active')).toBeTruthy();
+
+        spyOn(component, 'onTabActivated');
+
+        // programatically change tab by index
+        tabset.selectTab(2);
+
+        // check second tab is now active/first tab is not active
+        expect(tab3.classList.contains('active')).toBeTruthy();
+        expect(tab1.classList.contains('active')).toBeFalsy();
+
+        expect(component.onTabActivated).toHaveBeenCalled();
+    });
+
+    it('should change to new tab when tab is programmatically selected by tab instance', () => {
+
+        const tab1 = getTabForClass(0, nativeElement);
+        const tab3 = getTabForClass(2, nativeElement);
+
+        // check first tab is active
+        expect(tab1.classList.contains('active')).toBeTruthy();
+
+        spyOn(component, 'onTabActivated');
+
+        // programatically change tab by instance
+        tabset.selectTab(tabset._tabset.tabs[2]);
+
+        // check second tab is now active/first tab is not active
+        expect(tab3.classList.contains('active')).toBeTruthy();
+        expect(tab1.classList.contains('active')).toBeFalsy();
+
+        expect(component.onTabActivated).toHaveBeenCalled();
+    });
+
 });
 
 function getTab(index: number, nativeElement: HTMLElement): HTMLElement {
     return nativeElement.querySelectorAll<HTMLAnchorElement>('ux-tabset ul .nav-link').item(index);
 }
 
+function getTabForClass(index: number, nativeElement: HTMLElement): HTMLElement {
```

Comment:
This function appears to be identical to `getTab` - can it be removed?

Reply:
The 'active' class is applied at the li/.nav-item  level while getTab which is needed for other tests gets a tab at the a/.nav-link level.

Reply:
Ah sorry, I didn't catch that. Can we rename them to getTabLink/getTabItem to make it a bit more clear please?

Reply:
Will do